rachelnavarro/step/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,112 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
+  private ArrayList<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+  private int currentTime = 0;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Zero out possibleTimes and currentTime with every call to query().
+    possibleTimes = new ArrayList<TimeRange>();
+    currentTime = 0;
+
+    // Get prescheduled events and data about the meeting request.
+    long meetingLength = request.getDuration();
+    ArrayList<Event> prescheduledEvents = new ArrayList<Event>(events);
+
+    // A meeting with no atendees(and no optional attendees) can be scheduled for any time of the day.
+    if (request.getAttendees().size() == 0 && request.getOptionalAttendees().size() == 0) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    // A meeting that has a negative duration or that is longer than the whole day cannot be scheduled at all.
+    if (meetingLength < 0 || meetingLength > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+
+    // Sort the prescheduled events by their start time.
+    Collections.sort(prescheduledEvents, sortByStart);
+
+    // Determine available times and update possibleTimes based off of findings.
+    getPossibleTimes(prescheduledEvents, request.getAttendees(), meetingLength);
+    return possibleTimes;
+  }
+
+  /*
+   * Sort events by their start times.
+   */
+  private Comparator<Event> sortByStart =  (Event eventA, Event eventB) -> Long.compare(eventA.getWhen().start(), eventB.getWhen().start());
```

Comment:
If this is only intended on being used in the above method, remove this and place it in the .sort() call.

Reply:
Fixed! @dchin2